File: src/main/scala/edu/hitsz/application/HeroController.scala
package edu.hitsz.application

import edu.hitsz.aircraft.HeroAircraft

import java.awt.event.{MouseAdapter, MouseEvent}

/**
 * 英雄机控制类
 * 监听鼠标，控制英雄机的移动
 *
 * @author chiro2001
 */
class HeroController(var game: Game, var heroAircraft: HeroAircraft) {
  private val mouseAdapter = new MouseAdapter() {
    override def mouseDragged(e: MouseEvent): Unit = {
      super.mouseDragged(e)
      val x = e.getX
      val y = e.getY
      if (x < 0 || x > Main.WINDOW_WIDTH || y < 0 || y > Main.WINDOW_HEIGHT) { // 防止超出边界
        return
      }
      heroAircraft.setLocation(x, y)
    }
  }
  game.addMouseListener(mouseAdapter)
  game.addMouseMotionListener(mouseAdapter)
}

File: base/src/main/java/edu/hitsz/application/HeroController.java
package edu.hitsz.application;

import edu.hitsz.aircraft.HeroAircraft;
import edu.hitsz.aircraft.HeroAircraftFactory;

import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;

/**
 * 英雄机控制类
 * 监听鼠标，控制英雄机的移动
 *
 * @author hitsz
 */
public class HeroController {

    public HeroController(Game game) {

        MouseAdapter mouseAdapter = new MouseAdapter() {
            @Override
            public void mouseDragged(MouseEvent e) {
                super.mouseDragged(e);
                int x = e.getX();
                int y = e.getY();
                if (x < 0) {
                    x = 0;
                }
                if (x > Main.WINDOW_WIDTH) {
                    x = Main.WINDOW_WIDTH;
                }
                if (y < 0) {
                    y = 0;
                }
                if (y > Main.WINDOW_HEIGHT) {
                    y = Main.WINDOW_HEIGHT;
                }
                HeroAircraft heroAircraft = HeroAircraftFactory.getInstance();
                if (heroAircraft != null) {
                    heroAircraft.setPosition(x, y);
                }
            }
        };

        game.addMouseListener(mouseAdapter);
        game.addMouseMotionListener(mouseAdapter);
    }


}
